tesseract_core/sdk/cli.py

```diff
@@ -174,13 +176,18 @@ def main_callback(
 
     set_logger(loglevel, catch_warnings=True, rich_format=True)
 
+    try:
+        get_config()
+    except (FileNotFoundError, PermissionError, PydanticValidationError) as err:
```

Comment:
This function should only see generic validation errors, not specific ones.

```suggestion
    except PydanticValidationError as err:
```

Reply:
With your suggestion:

```console
❯ TESSERACT_DOCKER_EXECUTABLE="podman-hpc" uv run tesseract ps
 [x] Uncaught error
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /home/h/projects/pasteur/tesseract-core/tesseract_core/sdk/cli.py:180 in main_callback           │
│                                                                                                  │
│   177 │   set_logger(loglevel, catch_warnings=True, rich_format=True)                            │
│   178 │                                                                                          │
│   179 │   try:                                                                                   │
│ ❱ 180 │   │   get_config()                                                                       │
│   181 │   except PydanticValidationError as err:                                                 │
│   182 │   │   raise UserError(f"{err}") from None                                                │
│   183                                                                                            │
│                                                                                                  │
│ /home/h/projects/pasteur/tesseract-core/tesseract_core/sdk/config.py:84 in get_config            │
│                                                                                                  │
│   81 def get_config() -> RuntimeConfig:                                                          │
│   82 │   """Return the current runtime configuration."""                                         │
│   83 │   if _current_config is None:                                                             │
│ ❱ 84 │   │   update_config()                                                                     │
│   85 │                                                                                           │
│   86 │   return _current_config                                                                  │
│   87                                                                                             │
│                                                                                                  │
│ /home/h/projects/pasteur/tesseract-core/tesseract_core/sdk/config.py:74 in update_config         │
│                                                                                                  │
│   71 │                                                                                           │
│   72 │   conf_settings.update(kwargs)                                                            │
│   73 │                                                                                           │
│ ❱ 74 │   config = RuntimeConfig(**conf_settings)                                                 │
│   75 │   _current_config = config                                                                │
│   76                                                                                             │
│   77                                                                                             │
│                                                                                                  │
│ /home/h/projects/pasteur/tesseract-core/.venv/lib/python3.13/site-packages/pydantic/main.py:243  │
│ in __init__                                                                                      │
│                                                                                                  │
│    240 │   │   """                                                                               │
│    241 │   │   # `__tracebackhide__` tells pytest and some other tools to omit this function fr  │
│    242 │   │   __tracebackhide__ = True                                                          │
│ ❱  243 │   │   validated_self = self.__pydantic_validator__.validate_python(data, self_instance  │
│    244 │   │   if self is not validated_self:                                                    │
│    245 │   │   │   warnings.warn(                                                                │
│    246 │   │   │   │   'A custom validator is returning a value other than `self`.\n'            │
│                                                                                                  │
│ /home/h/projects/pasteur/tesseract-core/tesseract_core/sdk/config.py:23 in validate_executable   │
│                                                                                                  │
│   20 │                                                                                           │
│   21 │   exe = shutil.which(exe)                                                                 │
│   22 │   if exe is None:                                                                         │
│ ❱ 23 │   │   raise FileNotFoundError(f"Executable `{value}` not found.")                         │
│   24 │                                                                                           │
│   25 │   exe_path = Path(exe)                                                                    │
│   26 │   if not exe_path.is_file():                                                              │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
FileNotFoundError: Executable `podman-hpc` not found.
 [x] Aborting
```

> This function should only see generic validation errors, not specific ones.

Why?

Reply:
The issue was that pydantic only catches `ValueError` (and reformats them into `ValidationError`). Fixed now.